Formula/ntopng.rb

```diff
@@ -1,9 +1,18 @@
 class Ntopng < Formula
   desc "Next generation version of the original ntop"
   homepage "http://www.ntop.org/products/ntop/"
-  url "https://downloads.sourceforge.net/project/ntop/ntopng/ntopng-2.2.tar.gz"
-  sha256 "4fccfc9e9f333addcd3c957b4520c471117bc2df5655d6eabf328c7385fb255e"
-  revision 1
+
+  stable do
+    url "https://github.com/ntop/ntopng/archive/2.4.tar.gz"
+    sha256 "86f8ed46983f46bcd931304d3d992fc1af572b11e461ab9fb4f0f472429bd5dd"
+
+    resource "nDPI" do
+      # tip of 1.8-stable branch; four commits beyond the 1.8 tag
+      url "https://github.com/ntop/nDPI.git",
+        :tag => "1.8-stable",
```

Comment:
Yep, seems to be no issue there.

So I think in this case, the resource should be
```
resource "nDPI" do
  # 1.8-stable branch
  url "https://github.com/ntop/nDPI.git",
  :revision => "6fb81f146e2542cfbf7fab7d53678339c7747b35"
end 
```

So I think
a) we should always confirm a tag is a tag
b) we should always confirm a branch is a branch
c) we should always rev-parse matches :revision
d) we should always raise an error when (a branch is specified) && (a revision is specified || a tag is specified). Or perhaps (equivalently?) raise an error when a branch is used outside of a head spec.

So, for example, this appears to be bogus: https://github.com/Homebrew/homebrew-games/blob/master/ufoai.rb#L4